libraries/write_chef_error_handler.rb

```diff
@@ -0,0 +1,99 @@
+# frozen_string_literal: true
+
+#
+# Cookbook:: aws-parallelcluster
+#
+# Copyright:: 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with the
+# License. A copy of the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "LICENSE.txt" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES
+# OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions and
+# limitations under the License.
+
+module WriteChefError
+  # this class is used to handle chef errors and write the errors into a certain file if the file does not exist yet
+  class WriteChefError < Chef::Handler
+    def report
+      extend Chef::Mixin::ShellOut
```

Comment:
Why do we need to add `extend Chef::Mixin::ShellOut` here?

Reply:
In the previous PR, I used `Mixlib::ShellOut.new(xxx).run_command` at the beginning but was required to change to `shell_out` instead. To use the `shell_out` function, `extend Chef::Mixin::ShellOut` is needed.